saleor/order/tasks.py

```diff
@@ -21,3 +32,58 @@ def send_order_updated(order_ids):
     manager = get_plugins_manager()
     for order in Order.objects.filter(id__in=order_ids):
         manager.order_updated(order)
+
+
+def _batch_ids(iterable, batch_size=1):
+    length = len(iterable)
+    for index in range(0, length, batch_size):
+        yield iterable[index : min(index + batch_size, length)]
+
+
+def _queryset_in_batches(queryset):
+    ids = queryset.values_list("id", flat=True)
+    for ids_batch in _batch_ids(ids, 1000):
+        yield ids_batch
+
+
+def _expire_orders(manager, now):
+    qs = Order.objects.filter(
+        status=OrderStatus.UNCONFIRMED,
+        channel__expire_orders_after__isnull=False,
+        total_charged_amount=Decimal(0),
+    ).select_related("channel")
+    for ids_batch in _queryset_in_batches(qs):
+        with traced_atomic_transaction():
+            orders = Order.objects.filter(id__in=ids_batch).prefetch_related("user")
+            orders_to_update = []
+            for order in orders:
+                if _process_order(order, manager, now):
+                    orders_to_update.append(order)
+            Order.objects.bulk_update(orders_to_update, ["status", "updated_at"])
+
+
+def _process_order(order, manager, now):
+    if (
+        order.channel.expire_orders_after is not None
```

Comment:
What will happen If i provide expire_orders_after < 0 or expire_orders_after == 0?

Reply:
In both scenarios, the order will be cancelled the moment the job is triggered.

Reply:
Changed:
Values <0 are now not valid, and 0 will disable expiration of orders (the same as null value would do)

Reply:
FYI. We have the same strategy for reservations.